src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/controller/AssessmentController.kt

```diff
@@ -79,4 +109,225 @@ class AssessmentController(
+    @Parameter(description = "Assessment UUID", required = true, example = "123e4567-e89b-12d3-a456-426614174000")
+    @PathVariable
+    assessmentUuid: UUID,
+    @RequestBody @Valid
+    request: RollbackAssessmentRequest,
+  ): AssessmentResponse {
+    return AssessmentVersionCriteria(assessmentUuid, versionNumber = request.versionNumber)
+      .run(assessmentVersionService::find)
+      .let { assessmentVersionService.rollback(it, UserDetails.from(request)) }
+      .run(AssessmentResponse::from)
+  }
+
+  @RequestMapping(path = ["/{assessmentUuid}/soft-delete"], method = [RequestMethod.POST])
+  @Operation(description = "Soft-deletes an assessment.")
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "200", description = "Assessment has been soft-deleted"),
+      ApiResponse(
+        responseCode = "404",
+        description = "Assessment not found",
+        content = arrayOf(Content(schema = Schema(implementation = ErrorResponse::class))),
+      ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Unable to soft-delete an assessment that has already been soft-deleted",
+        content = arrayOf(Content(schema = Schema(implementation = ErrorResponse::class))),
+      ),
+      ApiResponse(
+        responseCode = "500",
+        description = "Unexpected error",
+        content = arrayOf(Content(schema = Schema(implementation = ErrorResponse::class))),
+      ),
+    ],
+  )
+  @PreAuthorize("hasAnyRole('ROLE_STRENGTHS_AND_NEEDS_OASYS', 'ROLE_STRENGTHS_AND_NEEDS_WRITE')")
+  fun softDelete(
+    @Parameter(description = "Assessment UUID", required = true, example = "123e4567-e89b-12d3-a456-426614174000")
+    @PathVariable
+    assessmentUuid: UUID,
+    @RequestBody @Valid
+    request: AuditedRequest,
+  ): Message {
+//    TODO: implement
+//    return assessmentService.softDelete(assessmentUuid, UserDetails.from(request))
+//      .run { Message("Successfully soft-deleted assessment UUID $assessmentUuid") }
+    return Message("Soft-deletion is not implemented")
+  }
+
+  @RequestMapping(path = ["/{assessmentUuid}/undelete"], method = [RequestMethod.POST])
+  @Operation(description = "Undeletes an OASys assessment.")
+  @ApiResponses(
+    value = [
+      ApiResponse(responseCode = "200", description = "OASys assessment has been undeleted"),
+      ApiResponse(
+        responseCode = "404",
+        description = "Assessment not found",
+        content = arrayOf(Content(schema = Schema(implementation = ErrorResponse::class))),
+      ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Unable to undelete an assessment that is not deleted",
+        content = arrayOf(Content(schema = Schema(implementation = ErrorResponse::class))),
+      ),
+      ApiResponse(
+        responseCode = "500",
+        description = "Unexpected error",
+        content = arrayOf(Content(schema = Schema(implementation = ErrorResponse::class))),
+      ),
+    ],
+  )
```

Comment:
Don't think you'll need to implement these bud. We'll do the `soft-delete`/`undelete` at the association level in the coordinator service 😃

Reply:
We thought so, but had some concerns about not letting SAN know what has been deleted. Let's catch up after standup